master/internal/job/job.go

```diff
@@ -2,111 +2,103 @@ package job
 
 import (
 	"fmt"
-	"slices"
 	"sort"
 	"strings"
 	"sync"
 
-	"github.com/pkg/errors"
 	"github.com/sirupsen/logrus"
+	"golang.org/x/exp/slices"
 
 	"github.com/determined-ai/determined/master/internal/rm"
 	"github.com/determined-ai/determined/master/internal/rm/rmerrors"
 	"github.com/determined-ai/determined/master/internal/sproto"
-	"github.com/determined-ai/determined/master/pkg/actor"
 	"github.com/determined-ai/determined/master/pkg/model"
 	"github.com/determined-ai/determined/proto/pkg/jobv1"
 )
 
-// Manager manages jobs.
-type Manager struct {
-	mu        sync.Mutex
-	rm        rm.ResourceManager
-	actorByID map[model.JobID]*actor.Ref
-	system    *actor.System
-	syslog    *logrus.Entry
+// Job is the interface for commands/experiments types that implement
+// the job service.
+type Job interface {
+	ToV1Job() *jobv1.Job
+	SetJobPriority(priority int) error
+	SetWeight(weight float64) error
 }
 
-// NewManager creates a new jobs manager instance.
-func NewManager(rm rm.ResourceManager, system *actor.System) *Manager {
-	return &Manager{
-		rm:        rm,
-		actorByID: make(map[model.JobID]*actor.Ref),
-		system:    system,
-		syslog:    logrus.WithField("component", "jobs"),
-	}
+// Service manages the job service.
+type Service struct {
+	mu          sync.Mutex
+	rm          rm.ResourceManager
+	genericByID map[model.JobID]Job
```

Comment:
```suggestion
	jobByID map[model.JobID]Job
```
nit: not sure if this is a better name